### ISSUE TITLE
Having .eslintrc in the root dir causes eslint tests to fail

### ISSUE THREAD

**Issue by fatfisz** [eslint/eslint#6206]
**What version of ESLint are you using?**
2.10.2
**What did you do?**
I cloned the `eslint` repo and ran tests.
**What did you expect to happen?**
No errors.
**What actually happened?**
A few errors (about 70 or so).

I think I might be the only one that has such a troubling configuration, because I've already created a few PRs to other quite popular repos, which added `"root": true` to the `.eslintrc` config file.

The problem is this: I have a "global" `.eslintrc` file that sits in the root dir. So every project that uses `eslint` and doesn't have `"root": true` set is also using my "global" config. It turns out that `eslint` itself has this problem, or rather `eslint` tests :smile: 

Changing the name of my "global" config file to something random solved the problem (all tests passed).

The simplest way to reproduce this is to create an `.eslintrc` file in the root dir with this content:
```
{
  "env": {
    "es6": true
  }
}
```

For me this caused 13 errors to emerge, for example:
```
  1) Config findLocalConfigFiles() should return an empty array when an .eslintrc file is not found:
     AssertionError: expected [ 'D:\\.eslintrc' ] to have a length of 0 but got 1
```
and
```
  6) Config getConfig() should merge configs when local .eslintrc overrides parent .eslintrc:

      AssertionError: expected { es6: true, node: true } to deeply equal { node: true }
      + expected - actual

       {
      -  "es6": true
         "node": true
       }
```

**Comment by nzakas** [comment]
Can you share all the errors you're seeing?

**Comment by fatfisz** [comment]
Right - I will paste the short 13-error version then:
```
  1) Config findLocalConfigFiles() should return an empty array when an .eslintrc file is not found:
     AssertionError: expected [ 'D:\\.eslintrc' ] to have a length of 0 but got 1
      at Function.assert.lengthOf (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:1086:37)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:150:20)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  2) Config findLocalConfigFiles() should return package.json only when no other config files are found:
     AssertionError: expected [ Array(3) ] to have a length of 2 but got 3
      at Function.assert.lengthOf (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:1086:37)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:160:20)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  3) Config findLocalConfigFiles() should return the only one config file even if there are multiple found:
     AssertionError: expected 2 to equal 1
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:172:20)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  4) Config findLocalConfigFiles() should return all possible files when multiple are found:

      AssertionError: expected [ Array(4) ] to deeply equal [ Array(3) ]
      + expected - actual

       [
         "D:\\Temp\\eslint\\fixtures\\config-hierarchy\\fileexts\\subdir\\subsubdir\\.eslintrc.json"
         "D:\\Temp\\eslint\\fixtures\\config-hierarchy\\fileexts\\subdir\\.eslintrc.yml"
         "D:\\Temp\\eslint\\fixtures\\config-hierarchy\\fileexts\\.eslintrc.js"
      -  "D:\\.eslintrc"
       ]

      at Function.assert.deepEqual (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:216:32)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:186:20)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  5) Config findLocalConfigFiles() should return an empty array when a package.json file is not found:
     AssertionError: expected [ 'D:\\.eslintrc' ] to have a length of 0 but got 1
      at Function.assert.lengthOf (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:1086:37)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:194:20)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  6) Config getConfig() should merge configs when local .eslintrc overrides parent .eslintrc:

      AssertionError: expected { es6: true, node: true } to deeply equal { node: true }
      + expected - actual

       {
      -  "es6": true
         "node": true
       }

      at Function.assert.deepEqual (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:216:32)
      at assertConfigsEqual (D:\Projects\eslint\tests\lib\config.js:61:16)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:445:13)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  7) Config getConfig() should merge configs when local .eslintrc overrides parent and grandparent .eslintrc:

      AssertionError: expected { es6: true, node: true } to deeply equal { node: true }
      + expected - actual

       {
      -  "es6": true
         "node": true
       }

      at Function.assert.deepEqual (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:216:32)
      at assertConfigsEqual (D:\Projects\eslint\tests\lib\config.js:61:16)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:466:13)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  8) Config getConfig() should merge command line config when config file adds to local .eslintrc:

      AssertionError: expected { es6: true, node: true } to deeply equal { node: true }
      + expected - actual

       {
      -  "es6": true
         "node": true
       }

      at Function.assert.deepEqual (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:216:32)
      at assertConfigsEqual (D:\Projects\eslint\tests\lib\config.js:61:16)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:504:13)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  9) Config getConfig() should merge command line config when config file overrides local .eslintrc:

      AssertionError: expected { es6: true, node: true } to deeply equal { node: true }
      + expected - actual

       {
      -  "es6": true
         "node": true
       }

      at Function.assert.deepEqual (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:216:32)
      at assertConfigsEqual (D:\Projects\eslint\tests\lib\config.js:61:16)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:527:13)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  10) Config getConfig() should merge command line config when config file adds to local and parent .eslintrc:

      AssertionError: expected { es6: true, node: true } to deeply equal { node: true }
      + expected - actual

       {
      -  "es6": true
         "node": true
       }

      at Function.assert.deepEqual (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:216:32)
      at assertConfigsEqual (D:\Projects\eslint\tests\lib\config.js:61:16)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:552:13)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  11) Config getConfig() should merge command line config when config file overrides local and parent .eslintrc:

      AssertionError: expected { es6: true, node: true } to deeply equal { node: true }
      + expected - actual

       {
      -  "es6": true
         "node": true
       }

      at Function.assert.deepEqual (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:216:32)
      at assertConfigsEqual (D:\Projects\eslint\tests\lib\config.js:61:16)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:576:13)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  12) Config getConfig() should merge command line config and rule when rule and config file overrides local .eslintrc:

      AssertionError: expected { es6: true, node: true } to deeply equal { node: true }
      + expected - actual

       {
      -  "es6": true
         "node": true
       }

      at Function.assert.deepEqual (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:216:32)
      at assertConfigsEqual (D:\Projects\eslint\tests\lib\config.js:61:16)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:602:13)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)

  13) Config getConfig() with env in a child configuration file should overwrite parserOptions of the parent with env of the child:

      AssertionError: expected { es6: true, commonjs: true } to deeply equal { commonjs: true }
      + expected - actual

       {
         "commonjs": true
      -  "es6": true
       }

      at Function.assert.deepEqual (D:\Projects\eslint\node_modules\chai\lib\chai\interface\assert.js:216:32)
      at assertConfigsEqual (D:\Projects\eslint\tests\lib\config.js:61:16)
      at Context.<anonymous> (D:\Projects\eslint\tests\lib\config.js:732:17)
      at callFn (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (D:\Projects\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (D:\Projects\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\Projects\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\Projects\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (D:\Projects\eslint\node_modules\mocha\lib\runner.js:320:5)
```

**Comment by fatfisz** [comment]
As you can see, the errors can be grouped into these categories:
- expected the array containing `.eslintrc` files to have one element less
- expected the array containing `.eslintrc` files to match an array without the root config file (`D:\.eslintrc`)
- expected `parserOptions` not to contain the `"es6": true` rule

**Comment by fatfisz** [comment]
@nzakas I can also post the 70-error version if it will help.

**Comment by btmills** [comment]
While we should still add `root: true` to our config, I'm curious about your use case for this. Do you intentionally want to modify project configs that don't specify `root: true`, or are you really just looking to provide your own personal default for projects that don't specify a config at all? If all you're going for is a fallback config, you can move your `.eslintrc` to your user home directory. If ESLint doesn't find any other configs as it traverses up to the root directory, it will [grab a default `.eslintrc` from the user's home directory](http://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy). That would unbreak your workflow for projects that haven't yet added `root: true`, though at the cost of a quick PR to a new project every so often :smirk:

**Comment by fatfisz** [comment]
@btmills Oooh, I didn't know about that. Yes, I was using this as a fallback config for my new projects, thanks for the advice.

**Comment by ilyavolodin** [comment]
I still think the original issue is correct and we should add `root: true` to our config. I know I've ran into an issue when I created .eslintrc file for testing in the folder above eslint, and then all of my unit tests started failing within ESLint. Took me a little while to figure out what's going on.

**Comment by gyandeeps** [comment]
Our project `.eslintrc` has `root: true` in it. I think the test are failing because some of the test gets copied into the temp folder for execution and thats when it goes and hits the home directory .eslintrc' file. I think i have this right, but correct me if i am wrong.

**Comment by fatfisz** [comment]
@gyandeeps Yes, I think I saw some error reports coming from files contained in `D:\Temp`, which is my default temporary directory.

**Comment by gyandeeps** [comment]
When I tried on my machine (Windows 7), i got 11 errors but only 5 had error output :confused: 

```sh
  10707 passing (46s)
  11 failing

  1) Config findLocalConfigFiles() should return an empty array when an .eslintrc file is not found:
     AssertionError: expected [ 'C:\\Users\\gs025879\\.eslintrc' ] to have a length of 0 but got 1
      at Function.assert.lengthOf (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\chai\lib\chai\interface\assert.js:1086:37)
      at Context.<anonymous> (C:\Users\gs025879\Documents\webstrom\eslint\tests\lib\config.js:150:20)
      at callFn (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:422:10)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:342:14)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:320:5)

  2) Config findLocalConfigFiles() should return package.json only when no other config files are found:
     AssertionError: expected [ Array(3) ] to have a length of 2 but got 3
      at Function.assert.lengthOf (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\chai\lib\chai\interface\assert.js:1086:37)
      at Context.<anonymous> (C:\Users\gs025879\Documents\webstrom\eslint\tests\lib\config.js:160:20)
      at callFn (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:422:10)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:342:14)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:320:5)

  3) Config findLocalConfigFiles() should return the only one config file even if there are multiple found:
     AssertionError: expected 2 to equal 1
      at Context.<anonymous> (C:\Users\gs025879\Documents\webstrom\eslint\tests\lib\config.js:172:20)
      at callFn (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:422:10)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:342:14)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:320:5)

  4) Config findLocalConfigFiles() should return all possible files when multiple are found:

      AssertionError: expected [ Array(4) ] to deeply equal [ Array(3) ]
      + expected - actual

       [
         "C:\\Users\\gs025879\\AppData\\Local\\Temp\\eslint\\fixtures\\config-hierarchy\\fileexts\\subdir\\subsubdir\\.eslintrc.json"
         "C:\\Users\\gs025879\\AppData\\Local\\Temp\\eslint\\fixtures\\config-hierarchy\\fileexts\\subdir\\.eslintrc.yml"
         "C:\\Users\\gs025879\\AppData\\Local\\Temp\\eslint\\fixtures\\config-hierarchy\\fileexts\\.eslintrc.js"
      -  "C:\\Users\\gs025879\\.eslintrc"
       ]

      at Function.assert.deepEqual (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\chai\lib\chai\interface\assert.js:216:32)
      at Context.<anonymous> (C:\Users\gs025879\Documents\webstrom\eslint\tests\lib\config.js:186:20)
      at callFn (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:422:10)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:342:14)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:320:5)

  5) Config findLocalConfigFiles() should return an empty array when a package.json file is not found:
     AssertionError: expected [ 'C:\\Users\\gs025879\\.eslintrc' ] to have a length of 0 but got 1
      at Function.assert.lengthOf (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\chai\lib\chai\interface\assert.js:1086:37)
      at Context.<anonymous> (C:\Users\gs025879\Documents\webstrom\eslint\tests\lib\config.js:194:20)
      at callFn (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:315:21)
      at Test.Runnable.run (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:308:7)
      at Runner.runTest (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:422:10)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:533:12
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:342:14)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:320:5)
```

Used the same steps as @fatfisz mentioned above.

**Comment by nzakas** [comment]
Since the errors are limited to the `Config` file, we can probably just go through all the fixtures for `Config` and add `root:true`.

**Comment by gyandeeps** [comment]
I dont think just adding `root: true` would completely solve the problem. Function under testing is [`findLocalConfigFiles`](https://github.com/eslint/eslint/blob/master/lib/config.js#L295) whose job is to discover all the files with a particular file name from the a directory to all the way up to home/base directory. Since its just discovering the files up the chain and not looking inside the content of file, we need to figure out a different solution.

**Comment by nzakas** [comment]
Ah, good point.

**Comment by gyandeeps** [comment]
So far https://github.com/tschaub/mock-fs looks promising as we can fake the directory structure rather than using the real directory structure.

**Comment by kaicataldo** [comment]
mock-fs is great!

**Comment by nzakas** [comment]
If someone wants to give it a try, go for it!

I had a lot of trouble with mock-fs when I tried to use it in some tests previously, but it's entirely possible I didn't know what I was doing. :)

**Comment by gyandeeps** [comment]
Here I have a sample work done for `findLocalConfigFiles` tests inside config.
https://github.com/eslint/eslint/commit/1b7b823c50685827b0a290a16a9006a04234bdca

With this change now the unit test are not failing using the same scenario as described in this issue.
I think like this we can fake all our fixtures in the future. :smile: 
@eslint/eslint-team thoughts?

**Comment by nzakas** [comment]
Very cool! Is there a way to split up that giant blob of data? Maybe each test can register just what it needs?

**Comment by gyandeeps** [comment]
Yes, ideally i want to create a json file for each kind of fixture and keep the file their. So the one I have is the exact json version of our fixtures/config-hierarchy (physical version).
Since its going to be a massive overtaking to convert all the files, i would propose to move to mock-fs from here on. Also as we touch different parts then we can slowly convert them.

**Comment by nzakas** [comment]
I think we can probably start with just updating these few failing tests so we can unblock potential contributors. Agree that we should use mock-fs going forward.

**Comment by gyandeeps** [comment]
@fatfisz If you have time plz, can try changes from https://github.com/eslint/eslint/pull/6402 on your machine to see if all the test pass. I created the same scenario on my machine as you described and it worked for me now. Just want to double check with you Thanks

**Comment by fatfisz** [comment]
@gyandeeps Sure, I will probably do it over the weekend :)

**Comment by fatfisz** [comment]
@gyandeeps Tried it right now and it worked for me (so Windows
